app/src/features/rules/screens/rulesList/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx

```diff
@@ -127,6 +130,27 @@ const useRuleTableColumns = (options: Record<string, boolean>) => {
                   </div>
                 </Tooltip>
               ) : null}
+
+              <RuleTypesDropdown
+                analyticEventSource={SOURCE.RULE_GROUP}
+                onOptionClick={(ruleType) => {
+                  createRuleAction(ruleType, SOURCE.RULE_GROUP);
+                  setSearchParams((params) => {
+                    params.set("groupId", group.id);
+                    return params;
+                  });
```

Comment:
createRuleAction should support passing groupId as a param and it should handle the logic of navigating with setting the query params